crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -57,7 +57,56 @@ def prepare
   api_version "2.0"
   def services
     if request.post?
-      head :not_implemented
+
+      # prepare the scripts for various actions in one chef run
+      service_object = CrowbarService.new(Rails.logger)
+      service_object.prepare_nodes_for_os_upgrade
+
+      # initiate the services shutdown in all clusters
+      # These are the non essential services that could be shut down everywhere
+      # FIXME: non-HA case not being handled
+      NodeObject.find("pacemaker_founder:true").each do |node|
+        node.ssh_cmd("/usr/sbin/crowbar-shutdown-services-before-upgrade.sh")
+      end
+      # FIXME: dump database on database node - currently missing
+
+      head :ok
+    else
+      render json: [], status: :not_implemented
```

Comment:
can we list the services here that will be stopped? if we can't then we may drop this endpoint